File: Spring 2021/Weekly Challenge 1/Submissions/Tommy Tran/Fibonacci.cpp
/*  [Challenge] Fibonacci and Continued
    Created by : Tommy Tran / Python47(Discord)
    Date Created: 2/24/2020                     */

#include <iostream>
#include <cmath>
using namespace std;

int fibonacci(int n);
int odd_Fibonacci(int n);

int main() {

    cout << "Enter 1: Fibonacci Program\n"
        << "Enter 2: Fibonacci Continued\n \n";

    int choice = 0, n = 0;

    cout << "Choice: ";
    cin >> choice;
    switch (choice) {

    case 1:
        cout << "Fibonacci Program\n \n";

        cout << "Please enter a fibonacci position number: ";
        cin >> n;
        cout << "Fibonacci Number: " << fibonacci(n) << "\n";
        break;

    case 2:
        cout << "Fibonacci Continued (Sums all odd n between 0 to n)\n \n";

        cout << "Please enter a fibonacci position number: ";
        cin >> n;
        cout << "Fibonacci Sum: " << odd_Fibonacci(n) << "\n";
        break;
    }
    return 0;
}

 
int fibonacci(int n) {
    if (n == 0) {
        return 0;
    }
    else if (n == 1) {
        return 1;
    }
    else {
        return fibonacci(n - 1) + fibonacci(n - 2);
    }
}

int odd_Fibonacci(int n) {
    int previous = 0;
    int current = 1;
    int result = 0;
    while (current <= n) {
        if (current % 2 != 0) {
            result += current;
        }
        current += previous;
        previous = current - previous;
    }
    return result;
}

File: Spring 2021/Weekly Challenge 1/Submissions/Amr Ojjeh/Fibonacci.py
def fib(n):
    final = 0
    for i in fib_seq(n):
        final = i
    return final

def fib_seq(n):
    if n <= 1:
        yield n
    else:
        current = 1
        prev = 0
        yield current
        for i in range(n - 1):
            temp = current
            current = current + prev
            prev = temp
            yield current

print(f"fib(0) = {fib(0)}")
print(f"fib(1) = {fib(1)}")
print(f"fib(6) = {fib(6)}")

# Fibonacci Continued

n = int(input("Enter n: "))
s = 0
print("0", end="")
for i in fib_seq(n):
    if i % 2 != 0:
        print(f" + {i}", end="")
        s += i
print(f" = {s}")
